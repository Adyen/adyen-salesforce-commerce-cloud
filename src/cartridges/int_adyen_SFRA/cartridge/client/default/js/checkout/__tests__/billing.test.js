/**
 * @jest-environment jsdom
 */
const { methods } = require('../billing');

describe('Billing', () => {
  it('should append html to payment details', () => {
    document.body.innerHTML = `
      <div class="payment-details">
        <span>some_child</span>
      </div>
    `;

    const selectedPaymentInstruments = [
      {
        selectedAdyenPM: 'mocked_pm',
        selectedIssuerName: 'mocked_issuer_name',
        maskedCreditCardNumber: 'mocked_masked_cc',
        expirationMonth: 'mocked_expiration_moth',
        expirationYear: 'mocked_expiration_year',
      },
    ];
    const order = {
      resources: {
        cardEnding: 'mocked_card_ending',
      },
      billing: {
        payment: {
          selectedPaymentInstruments,
        },
      },
    };
    methods.updatePaymentInformation(order);
    expect(document.querySelector('.payment-details')).toMatchSnapshot();
  });
});
