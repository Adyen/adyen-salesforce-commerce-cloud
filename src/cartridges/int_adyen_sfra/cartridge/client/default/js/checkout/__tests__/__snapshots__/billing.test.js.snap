// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Billing should append html to payment details 1`] = `
<div
  class="payment-details"
>
  <div>
    <span>
      mocked_pm
    </span>
  </div>
  <div>
    <span>
      mocked_issuer_name
    </span>
  </div>
  <div>
    mocked_masked_cc
  </div>
  <div>
    <span>
      mocked_card_ending mocked_expiration_moth/mocked_expiration_year
    </span>
  </div>
</div>
`;
